internal/plugincommon/discovery/processor.go

```diff
@@ -53,19 +53,28 @@ func (cdp *ContractDiscoveryProcessor) Query(_ context.Context, _ dt.Outcome) (d
 func (cdp *ContractDiscoveryProcessor) Observation(
 	ctx context.Context, _ dt.Outcome, _ dt.Query,
 ) (dt.Observation, error) {
+	// all oracles should be able to read from the home chain, so we
+	// can fetch f chain reliably.
+	// TODO: should we error out here or just try to "observe everything we can"?
+	// i.e, similar to the commit plugin.
```

Comment:
@winder lmk your thoughts on this

Reply:
Let's error early and often, we can relax them later as we harden the system.

For fChain specifically, it's a problem. The current design requires observing it multiple times per round of observations.

The simplest way to address this would be adding fChain as a parameter to the `PluginProcesssor` type.

Reply:
Yeah actually you're right, fChain is pretty much a dependency for all upcoming consensus operations, so it should error.

Reply:
[Created a ticket for this.](https://smartcontract-it.atlassian.net/jira/software/c/projects/CCIP/boards/423?selectedIssue=CCIP-3526)